diff -aurN --ignore-all-space shibboleth-idp-3.3.3.orig/views/duo.vm shibboleth-idp-3.3.3.gluu/views/duo.vm
--- shibboleth-idp-3.3.3.orig/views/duo.vm	2018-05-15 06:49:38.000000000 +0300
+++ shibboleth-idp-3.3.3.gluu/views/duo.vm	2019-07-07 10:19:38.307911200 +0300
@@ -39,7 +39,7 @@
     <div class="wrapper">
       <div class="container">
         <header>
-          <img src="$request.getContextPath()#springMessage("idp.logo")" alt="#springMessageText("idp.logo.alt-text", "logo")">
+          <img src="$request.getContextPath()/servlet/logo" alt="#springMessageText("idp.logo.alt-text", "logo")">
         </header>
 
         <div class="content">
diff -aurN --ignore-all-space shibboleth-idp-3.3.3.orig/views/error.vm shibboleth-idp-3.3.3.gluu/views/error.vm
--- shibboleth-idp-3.3.3.orig/views/error.vm	2018-05-15 06:49:38.000000000 +0300
+++ shibboleth-idp-3.3.3.gluu/views/error.vm	2019-07-07 10:19:38.308908600 +0300
@@ -53,7 +53,7 @@
     <div class="wrapper">
     	<div class="container">
         	<header>
-				<img src="$request.getContextPath()#springMessage("idp.logo")" alt="#springMessageText("idp.logo.alt-text", "logo")">
+				<img src="$request.getContextPath()/servlet/logo" alt="#springMessageText("idp.logo.alt-text", "logo")">
 				<h3>$title - $titleSuffix</h3>
 			</header>
 		
diff -aurN --ignore-all-space shibboleth-idp-3.3.3.orig/views/login.vm shibboleth-idp-3.3.3.gluu/views/login.vm
--- shibboleth-idp-3.3.3.orig/views/login.vm	2018-05-15 06:49:38.000000000 +0300
+++ shibboleth-idp-3.3.3.gluu/views/login.vm	2019-07-07 10:19:38.308908600 +0300
@@ -38,7 +38,7 @@
     <div class="wrapper">
       <div class="container">
         <header>
-          <img src="$request.getContextPath()#springMessage("idp.logo")" alt="#springMessageText("idp.logo.alt-text", "logo")">
+          <img src="$request.getContextPath()/servlet/logo" alt="#springMessageText("idp.logo.alt-text", "logo")">
         </header>
 
         <div class="content">
diff -aurN --ignore-all-space shibboleth-idp-3.3.3.orig/views/logout-complete.vm shibboleth-idp-3.3.3.gluu/views/logout-complete.vm
--- shibboleth-idp-3.3.3.orig/views/logout-complete.vm	2018-05-15 06:49:38.000000000 +0300
+++ shibboleth-idp-3.3.3.gluu/views/logout-complete.vm	2019-07-07 10:19:38.309905900 +0300
@@ -27,7 +27,7 @@
     <div class="wrapper">
       <div class="container">
         <header>
-          <img src="$request.getContextPath()#springMessage("idp.logo")" alt="#springMessageText("idp.logo.alt-text", "logo")">
+          <img src="$request.getContextPath()/servlet/logo" alt="#springMessageText("idp.logo.alt-text", "logo")">
         </header>
 
         <div class="content">
diff -aurN --ignore-all-space shibboleth-idp-3.3.3.orig/views/logout-propagate.vm shibboleth-idp-3.3.3.gluu/views/logout-propagate.vm
--- shibboleth-idp-3.3.3.orig/views/logout-propagate.vm	2018-05-15 06:49:38.000000000 +0300
+++ shibboleth-idp-3.3.3.gluu/views/logout-propagate.vm	2019-07-07 10:19:38.309905900 +0300
@@ -30,7 +30,7 @@
     <div class="wrapper">
       <div class="container">
         <header>
-          <img src="$request.getContextPath()#springMessage("idp.logo")" alt="#springMessageText("idp.logo.alt-text", "logo")">
+          <img src="$request.getContextPath()/servlet/logo" alt="#springMessageText("idp.logo.alt-text", "logo")">
         </header>
 
         <div class="content">
diff -aurN --ignore-all-space shibboleth-idp-3.3.3.orig/views/logout.vm shibboleth-idp-3.3.3.gluu/views/logout.vm
--- shibboleth-idp-3.3.3.orig/views/logout.vm	2018-05-15 06:49:38.000000000 +0300
+++ shibboleth-idp-3.3.3.gluu/views/logout.vm	2019-07-07 10:19:38.310903400 +0300
@@ -34,7 +34,7 @@
     <div class="wrapper">
       <div class="container">
         <header>
-          <img src="$request.getContextPath()#springMessage("idp.logo")" alt="#springMessageText("idp.logo.alt-text", "logo")">
+          <img src="$request.getContextPath()/servlet/logo" alt="#springMessageText("idp.logo.alt-text", "logo")">
         </header>
 
         <div class="content">
diff -aurN --ignore-all-space shibboleth-idp-3.3.3.orig/views/spnego-unavailable.vm shibboleth-idp-3.3.3.gluu/views/spnego-unavailable.vm
--- shibboleth-idp-3.3.3.orig/views/spnego-unavailable.vm	2018-05-15 06:49:38.000000000 +0300
+++ shibboleth-idp-3.3.3.gluu/views/spnego-unavailable.vm	2019-07-07 10:19:38.311900600 +0300
@@ -25,7 +25,7 @@
     <div class="wrapper">
       <div class="container">
         <header>
-          <img src="$request.getContextPath()#springMessage("idp.logo")" alt="#springMessageText("idp.logo.alt-text", "logo")">
+          <img src="$request.getContextPath()/servlet/logo" alt="#springMessageText("idp.logo.alt-text", "logo")">
           <h3>#springMessageText("idp.title", "Web Login Sevice") - #springMessage("idp.title.suffix", "Error")</h3>
         </header>
                 
diff -aurN --ignore-all-space shibboleth-idp-3.3.3.orig/views/user-prefs.vm shibboleth-idp-3.3.3.gluu/views/user-prefs.vm
--- shibboleth-idp-3.3.3.orig/views/user-prefs.vm	2018-05-15 06:49:38.000000000 +0300
+++ shibboleth-idp-3.3.3.gluu/views/user-prefs.vm	2019-07-07 10:19:38.311900600 +0300
@@ -24,7 +24,7 @@
     <div class="wrapper">
       <div class="container">
         <header>
-          <img src="$request.getContextPath()#springMessage("idp.logo")" alt="#springMessageText("idp.logo.alt-text", "logo")">
+          <img src="$request.getContextPath()/servlet/logo" alt="#springMessageText("idp.logo.alt-text", "logo")">
           <h3>#springMessageText("idp.title", "Web Login Service") - #springMessageText("idp.userprefs.title.suffix", "Login Preferences")</h3>
           <p>
           #springMessage("idp.userprefs.info")
