diff --git a/src/main/java/net/shibboleth/utilities/java/support/collection/ClassIndexedSet.java b/src/main/java/net/shibboleth/utilities/java/support/collection/ClassIndexedSet.java
index 7b3fbdd..39b97f3 100644
--- a/src/main/java/net/shibboleth/utilities/java/support/collection/ClassIndexedSet.java
+++ b/src/main/java/net/shibboleth/utilities/java/support/collection/ClassIndexedSet.java
@@ -34,9 +34,11 @@ import net.shibboleth.utilities.java.support.logic.Constraint;
  * 
  * @param <T> the type of object stored by this class
  */
-public class ClassIndexedSet<T> extends AbstractSet<T> implements Set<T> {
+public class ClassIndexedSet<T> extends AbstractSet<T> implements Set<T>, java.io.Serializable {
 
-    /** Storage for set members. */
+	private static final long serialVersionUID = -4684766407987143505L;
+
+	/** Storage for set members. */
     private final HashSet<T> set;
 
     /** Storage for index of class -&gt; member. */
diff --git a/src/main/java/net/shibboleth/utilities/java/support/component/Component.java b/src/main/java/net/shibboleth/utilities/java/support/component/Component.java
index c5b0170..1ffac53 100644
--- a/src/main/java/net/shibboleth/utilities/java/support/component/Component.java
+++ b/src/main/java/net/shibboleth/utilities/java/support/component/Component.java
@@ -18,6 +18,6 @@
 package net.shibboleth.utilities.java.support.component;
 
 /** A marker interface that represents something used as a discrete part of a larger system. */
-public interface Component {
+public interface Component extends java.io.Serializable {
 
 }
\ No newline at end of file
