diff --git a/opensaml-parent/pom.xml b/opensaml-parent/pom.xml
index f1a1264d3..441f564c0 100644
--- a/opensaml-parent/pom.xml
+++ b/opensaml-parent/pom.xml
@@ -26,6 +26,14 @@
     -->
     <packaging>pom</packaging>
 
+    <repositories>
+        <repository>
+            <id>shib</id>
+            <name>Shib</name>
+            <url>https://build.shibboleth.net/nexus/content/repositories/releases/</url>
+        </repository>
+    </repositories>
+
     <modules>
         <module>../opensaml-core</module>
         <module>../opensaml-storage-api</module>
diff --git a/opensaml-core/src/main/java/org/opensaml/core/xml/Namespace.java b/opensaml-core/src/main/java/org/opensaml/core/xml/Namespace.java
index 2c9850562..4d6ac3d33 100644
--- a/opensaml-core/src/main/java/org/opensaml/core/xml/Namespace.java
+++ b/opensaml-core/src/main/java/org/opensaml/core/xml/Namespace.java
@@ -25,7 +25,7 @@ import net.shibboleth.utilities.java.support.primitive.StringSupport;
 import net.shibboleth.utilities.java.support.xml.XMLConstants;
 
 /** Data structure for representing XML namespace attributes. */
-public class Namespace {
+public class Namespace implements ava.io.Serializable {
 
     /** URI of the namespace. */
     private String namespaceURI;
diff --git a/opensaml-core/src/main/java/org/opensaml/core/xml/NamespaceManager.java b/opensaml-core/src/main/java/org/opensaml/core/xml/NamespaceManager.java
index cc6cc95c8..a52980eb6 100644
--- a/opensaml-core/src/main/java/org/opensaml/core/xml/NamespaceManager.java
+++ b/opensaml-core/src/main/java/org/opensaml/core/xml/NamespaceManager.java
@@ -44,7 +44,7 @@ import com.google.common.base.Strings;
  * should call the appropriate method, based on the type of change being made.
  * </p>
  */
-public class NamespaceManager {
+public class NamespaceManager implements java.io.Serializable {
     
     /** The token used to represent the default namespace in {@link #getNonVisibleNamespacePrefixes()}. */
     public static final String DEFAULT_NS_TOKEN = "#default";
diff --git a/opensaml-core/src/main/java/org/opensaml/core/xml/XMLObject.java b/opensaml-core/src/main/java/org/opensaml/core/xml/XMLObject.java
index bf9af8d54..a1f3bacd9 100644
--- a/opensaml-core/src/main/java/org/opensaml/core/xml/XMLObject.java
+++ b/opensaml-core/src/main/java/org/opensaml/core/xml/XMLObject.java
@@ -34,7 +34,7 @@ import org.w3c.dom.Element;
  * A object that represents an XML element, usually of a specific schema type, that has been unmarshalled into this Java
  * object.
  */
-public interface XMLObject {
+public interface XMLObject extends java.io.Serializable {
 
     /**
      * Detaches the XMLObject from its parent. This will release the parent's cached DOM (if it has one) and set this
diff --git a/opensaml-core/src/main/java/org/opensaml/core/xml/schema/XSBooleanValue.java b/opensaml-core/src/main/java/org/opensaml/core/xml/schema/XSBooleanValue.java
index 6b53a916e..608d14c45 100644
--- a/opensaml-core/src/main/java/org/opensaml/core/xml/schema/XSBooleanValue.java
+++ b/opensaml-core/src/main/java/org/opensaml/core/xml/schema/XSBooleanValue.java
@@ -25,7 +25,7 @@ import net.shibboleth.utilities.java.support.primitive.StringSupport;
  * A class representing a boolean attribute. This class tracks the usage of the literals {true, false, 1, 0} to ensure
  * proper roundtripping when unmarshalling/marshalling.
  */
-public class XSBooleanValue {
+public class XSBooleanValue implements java.io.Serializable {
 
     /** Whether to use the numeric representation of the lexical one. */
     private boolean numeric;
diff --git a/opensaml-core/src/main/java/org/opensaml/core/xml/util/AttributeMap.java b/opensaml-core/src/main/java/org/opensaml/core/xml/util/AttributeMap.java
index 3cba34288..aee7966d7 100644
--- a/opensaml-core/src/main/java/org/opensaml/core/xml/util/AttributeMap.java
+++ b/opensaml-core/src/main/java/org/opensaml/core/xml/util/AttributeMap.java
@@ -46,7 +46,7 @@ import com.google.common.base.Strings;
  * attributes change.
  */
 @NotThreadSafe
-public class AttributeMap implements Map<QName, String> {
+public class AttributeMap implements Map<QName, String>, java.io.Serializable {
     
     /** Logger. */
     private final Logger log = LoggerFactory.getLogger(AttributeMap.class);
diff --git a/opensaml-core/src/main/java/org/opensaml/core/xml/util/IDIndex.java b/opensaml-core/src/main/java/org/opensaml/core/xml/util/IDIndex.java
index 3822c60c7..aef5e9f8b 100644
--- a/opensaml-core/src/main/java/org/opensaml/core/xml/util/IDIndex.java
+++ b/opensaml-core/src/main/java/org/opensaml/core/xml/util/IDIndex.java
@@ -36,7 +36,7 @@ import org.opensaml.core.xml.XMLObject;
  * {@link org.opensaml.core.xml.XMLObject}.
  */
 @NotThreadSafe
-public class IDIndex {
+public class IDIndex implements java.io.Serializable {
     
     /** The XMLObject which owns this ID index. */
     @Nonnull private final XMLObject owner;
diff --git a/opensaml-core/src/main/java/org/opensaml/core/xml/util/XMLObjectChildrenList.java b/opensaml-core/src/main/java/org/opensaml/core/xml/util/XMLObjectChildrenList.java
index f03b880f1..a70405a83 100644
--- a/opensaml-core/src/main/java/org/opensaml/core/xml/util/XMLObjectChildrenList.java
+++ b/opensaml-core/src/main/java/org/opensaml/core/xml/util/XMLObjectChildrenList.java
@@ -39,7 +39,7 @@ import com.google.common.collect.Collections2;
  * 
  * @param <ElementType> type of elements added to the list
  */
-public class XMLObjectChildrenList<ElementType extends XMLObject> extends AbstractList<ElementType> {
+public class XMLObjectChildrenList<ElementType extends XMLObject> extends AbstractList<ElementType> implements java.io.Serializable {
 
     /** Parent to the elements in this list. */
     private final XMLObject parent;
diff --git a/opensaml-messaging-api/src/main/java/org/opensaml/messaging/context/BaseContext.java b/opensaml-messaging-api/src/main/java/org/opensaml/messaging/context/BaseContext.java
index f4c14e55f..898c43879 100644
--- a/opensaml-messaging-api/src/main/java/org/opensaml/messaging/context/BaseContext.java
+++ b/opensaml-messaging-api/src/main/java/org/opensaml/messaging/context/BaseContext.java
@@ -57,7 +57,7 @@ import org.slf4j.LoggerFactory;
  * to this convention, auto-creation will fail.
  * </p>
  */
-public abstract class BaseContext implements Iterable<BaseContext> {
+public abstract class BaseContext implements Iterable<BaseContext>, java.io.Serializable {
 
     /** Logger. */
     @Nonnull private final Logger log = LoggerFactory.getLogger(BaseContext.class);
diff --git a/opensaml-messaging-api/src/main/java/org/opensaml/messaging/context/navigate/ChildContextLookup.java b/opensaml-messaging-api/src/main/java/org/opensaml/messaging/context/navigate/ChildContextLookup.java
index 9f55efd6d..f6b88cef3 100644
--- a/opensaml-messaging-api/src/main/java/org/opensaml/messaging/context/navigate/ChildContextLookup.java
+++ b/opensaml-messaging-api/src/main/java/org/opensaml/messaging/context/navigate/ChildContextLookup.java
@@ -32,7 +32,7 @@ import net.shibboleth.utilities.java.support.logic.Constraint;
  * @param <ChildContext> type of the child context
  */
 public class ChildContextLookup<ParentContext extends BaseContext, ChildContext extends BaseContext> implements
-        ContextDataLookupFunction<ParentContext, ChildContext> {
+        ContextDataLookupFunction<ParentContext, ChildContext>, java.io.Serializable {
 
     /** Child context type to look up. */
     @Nonnull private final Class<ChildContext> childType;
diff --git a/opensaml-profile-api/src/main/java/org/opensaml/profile/logic/NoIntegrityMessageChannelPredicate.java b/opensaml-profile-api/src/main/java/org/opensaml/profile/logic/NoIntegrityMessageChannelPredicate.java
index da39c214e..9b4330c88 100644
--- a/opensaml-profile-api/src/main/java/org/opensaml/profile/logic/NoIntegrityMessageChannelPredicate.java
+++ b/opensaml-profile-api/src/main/java/org/opensaml/profile/logic/NoIntegrityMessageChannelPredicate.java
@@ -30,7 +30,7 @@ import net.shibboleth.utilities.java.support.logic.Predicate;
  * 
  * <p>Typically but not exclusively used as a predicate for whether to sign something.</p>
  */
-public class NoIntegrityMessageChannelPredicate implements Predicate<ProfileRequestContext> {
+public class NoIntegrityMessageChannelPredicate implements Predicate<ProfileRequestContext>, java.io.Serializable {
 
     /** {@inheritDoc} */
     public boolean test(@Nullable final ProfileRequestContext input) {
diff --git a/opensaml-saml-api/src/main/java/org/opensaml/saml/common/SAMLVersion.java b/opensaml-saml-api/src/main/java/org/opensaml/saml/common/SAMLVersion.java
index dce7c92ff..a9b32acb1 100644
--- a/opensaml-saml-api/src/main/java/org/opensaml/saml/common/SAMLVersion.java
+++ b/opensaml-saml-api/src/main/java/org/opensaml/saml/common/SAMLVersion.java
@@ -24,7 +24,7 @@ import net.shibboleth.utilities.java.support.annotation.constraint.NotEmpty;
 /**
  * A type safe SAML version enumeration.
  */
-public final class SAMLVersion {
+public final class SAMLVersion implements java.io.Serializable {
 
     /** SAML version 1.0. */
     @Nonnull public static final SAMLVersion VERSION_10 = new SAMLVersion(1, 0);
diff --git a/opensaml-saml-api/src/main/java/org/opensaml/saml/saml2/metadata/EntityDescriptor.java b/opensaml-saml-api/src/main/java/org/opensaml/saml/saml2/metadata/EntityDescriptor.java
index 209234119..c0d3b4714 100644
--- a/opensaml-saml-api/src/main/java/org/opensaml/saml/saml2/metadata/EntityDescriptor.java
+++ b/opensaml-saml-api/src/main/java/org/opensaml/saml/saml2/metadata/EntityDescriptor.java
@@ -31,7 +31,7 @@ import org.opensaml.saml.saml2.common.TimeBoundSAMLObject;
  * SAML 2.0 Metadata EntityDescriptor.
  */
 public interface EntityDescriptor extends SignableSAMLObject, TimeBoundSAMLObject, CacheableSAMLObject,
-        AttributeExtensibleXMLObject {
+        AttributeExtensibleXMLObject, java.io.Serializable {
 
     /** Element name, no namespace. */
     public static final String DEFAULT_ELEMENT_LOCAL_NAME = "EntityDescriptor";
diff --git a/opensaml-security-api/src/main/java/org/opensaml/security/credential/Credential.java b/opensaml-security-api/src/main/java/org/opensaml/security/credential/Credential.java
index a16044fb6..5022ec7b5 100644
--- a/opensaml-security-api/src/main/java/org/opensaml/security/credential/Credential.java
+++ b/opensaml-security-api/src/main/java/org/opensaml/security/credential/Credential.java
@@ -31,7 +31,7 @@ import javax.crypto.SecretKey;
  * With asymmetric key-based credentials, local entity credentials will usually contain both a public 
  * and private key while peer credentials will normally contain only a public key. 
  */
-public interface Credential {
+public interface Credential extends java.io.Serializable {
     
     /**
      * The unique ID of the entity this credential is for.
diff --git a/opensaml-security-api/src/main/java/org/opensaml/security/x509/InternalX500DNHandler.java b/opensaml-security-api/src/main/java/org/opensaml/security/x509/InternalX500DNHandler.java
index 7bae2a3bf..65dde473f 100644
--- a/opensaml-security-api/src/main/java/org/opensaml/security/x509/InternalX500DNHandler.java
+++ b/opensaml-security-api/src/main/java/org/opensaml/security/x509/InternalX500DNHandler.java
@@ -26,7 +26,7 @@ import net.shibboleth.utilities.java.support.logic.Constraint;
  * Basic implementation of {@link X500DNHandler} which uses the internal built-in mechanisms
  * provided by {@link X500Principal} directly.
  */
-public class InternalX500DNHandler implements X500DNHandler {
+public class InternalX500DNHandler implements X500DNHandler, java.io.Serializable {
 
     /** {@inheritDoc} */
     @Nonnull public byte[] getEncoded(@Nonnull final X500Principal principal) {
diff --git a/opensaml-security-api/src/main/java/org/opensaml/security/x509/tls/ClientTLSValidationParameters.java b/opensaml-security-api/src/main/java/org/opensaml/security/x509/tls/ClientTLSValidationParameters.java
index 630828ae4..41531026a 100644
--- a/opensaml-security-api/src/main/java/org/opensaml/security/x509/tls/ClientTLSValidationParameters.java
+++ b/opensaml-security-api/src/main/java/org/opensaml/security/x509/tls/ClientTLSValidationParameters.java
@@ -25,7 +25,7 @@ import org.opensaml.security.x509.X509Credential;
 /**
  * Parameters used in validating a client TLS {@link X509Credential}.
  */
-public class ClientTLSValidationParameters {
+public class ClientTLSValidationParameters implements java.io.Serializable {
     
     /** A {@link TrustEngine} instance used to validate a client TLS {@link X509Credential}. **/
     @Nullable private TrustEngine<? super X509Credential> x509TrustEngine;
diff --git a/opensaml-storage-impl/.gitignore b/opensaml-storage-impl/.gitignore
index 1df25bfc8..a5e64b76f 100644
--- a/opensaml-storage-impl/.gitignore
+++ b/opensaml-storage-impl/.gitignore
@@ -1,2 +1,3 @@
 /test-output
 /target
+/bin/
diff --git a/opensaml-storage-impl/src/main/java/org/opensaml/storage/impl/client/JSONClientStorageServiceStore.java b/opensaml-storage-impl/src/main/java/org/opensaml/storage/impl/client/JSONClientStorageServiceStore.java
index 17c828219..ba40fefbb 100644
--- a/opensaml-storage-impl/src/main/java/org/opensaml/storage/impl/client/JSONClientStorageServiceStore.java
+++ b/opensaml-storage-impl/src/main/java/org/opensaml/storage/impl/client/JSONClientStorageServiceStore.java
@@ -46,7 +46,7 @@ import net.shibboleth.utilities.java.support.security.DataSealerException;
 /**
  * JSON-based storage for {@link ClientStorageService}.
  */
-public class JSONClientStorageServiceStore extends AbstractClientStorageServiceStore {
+public class JSONClientStorageServiceStore extends AbstractClientStorageServiceStore implements java.io.Serializable {
 
     /** Class logger. */
     @Nonnull private final Logger log = LoggerFactory.getLogger(JSONClientStorageServiceStore.class);
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/AlgorithmPolicyConfiguration.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/AlgorithmPolicyConfiguration.java
index 421eb8968..6f3935e85 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/AlgorithmPolicyConfiguration.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/AlgorithmPolicyConfiguration.java
@@ -28,7 +28,7 @@ import net.shibboleth.utilities.java.support.annotation.constraint.Unmodifiable;
 /**
  * Algorithm URI include/exclude policy configuration.
  */
-public interface AlgorithmPolicyConfiguration {
+public interface AlgorithmPolicyConfiguration extends java.io.Serializable {
     
     /** Rule precedence values. */
     public enum Precedence {
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/AlgorithmPolicyParameters.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/AlgorithmPolicyParameters.java
index 481e8a88f..6b69568f7 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/AlgorithmPolicyParameters.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/AlgorithmPolicyParameters.java
@@ -32,7 +32,7 @@ import net.shibboleth.utilities.java.support.primitive.StringSupport;
 /**
  * The algorithm policy parameters.
  */
-public class AlgorithmPolicyParameters {
+public class AlgorithmPolicyParameters implements java.io.Serializable {
     
     /** Included algorithm URIs. */
     @Nonnull @NonnullElements private Collection<String> includedAlgorithmURIs;
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionConfiguration.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionConfiguration.java
index 26af52d9f..7d8dbcfa0 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionConfiguration.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionConfiguration.java
@@ -26,7 +26,7 @@ import org.opensaml.xmlsec.keyinfo.KeyInfoCredentialResolver;
  * The configuration information to use when decrypting encrypted XML.
  */
 @SuppressWarnings("removal")
-public interface DecryptionConfiguration extends WhitelistBlacklistConfiguration {
+public interface DecryptionConfiguration extends WhitelistBlacklistConfiguration, java.io.Serializable {
     
     /**
      * The KeyInfoCredentialResolver to use when processing the EncryptedData/KeyInfo.
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionParameters.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionParameters.java
index 402a86d0d..c0813245c 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionParameters.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionParameters.java
@@ -26,7 +26,7 @@ import org.opensaml.xmlsec.keyinfo.KeyInfoCredentialResolver;
  * The effective parameters to use when decrypting encrypted XML.
  */
 @SuppressWarnings("removal")
-public class DecryptionParameters extends WhitelistBlacklistParameters {
+public class DecryptionParameters extends WhitelistBlacklistParameters implements java.io.Serializable {
     
     /** The EncryptedData's KeyInfo credential resolver. */
     @Nullable private KeyInfoCredentialResolver dataKeyInfoCredentialResolver;
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionParametersResolver.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionParametersResolver.java
index 88f099a8d..a74b5c418 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionParametersResolver.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/DecryptionParametersResolver.java
@@ -23,6 +23,6 @@ import net.shibboleth.utilities.java.support.resolver.Resolver;
 /**
  * An interface for components which resolve {@link DecryptionParameters} based on a {@link CriteriaSet}.
  */
-public interface DecryptionParametersResolver extends Resolver<DecryptionParameters, CriteriaSet> {
+public interface DecryptionParametersResolver extends Resolver<DecryptionParameters, CriteriaSet>, java.io.Serializable {
 
 }
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionConfiguration.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionConfiguration.java
index 5a70942c7..1127779c3 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionConfiguration.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionConfiguration.java
@@ -37,7 +37,7 @@ import org.opensaml.xmlsec.keyinfo.NamedKeyInfoGeneratorManager;
  * The configuration information to use when generating encrypted XML.
  */
 @SuppressWarnings("removal")
-public interface EncryptionConfiguration extends WhitelistBlacklistConfiguration {
+public interface EncryptionConfiguration extends WhitelistBlacklistConfiguration, java.io.Serializable {
     
     /**
      * Get the list of data encryption credentials to use, in preference order.
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionParameters.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionParameters.java
index 2d5f4457b..a947ab5d9 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionParameters.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionParameters.java
@@ -27,7 +27,7 @@ import org.opensaml.xmlsec.keyinfo.KeyInfoGenerator;
 /**
  * The effective parameters to use when generating encrypted XML.
  */
-public class EncryptionParameters {
+public class EncryptionParameters implements java.io.Serializable {
     
     /** The EncryptedData encryption credential. */
     private Credential dataEncryptionCredential;
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionParametersResolver.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionParametersResolver.java
index 90595b1b8..9531a3403 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionParametersResolver.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/EncryptionParametersResolver.java
@@ -23,6 +23,6 @@ import net.shibboleth.utilities.java.support.resolver.Resolver;
 /**
  * An interface for components which resolve {@link EncryptionParameters} based on a {@link CriteriaSet}.
  */
-public interface EncryptionParametersResolver extends Resolver<EncryptionParameters, CriteriaSet> {
+public interface EncryptionParametersResolver extends Resolver<EncryptionParameters, CriteriaSet>, java.io.Serializable {
 
 }
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/KeyTransportAlgorithmPredicate.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/KeyTransportAlgorithmPredicate.java
index b3c7b8ba9..b2ed11d79 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/KeyTransportAlgorithmPredicate.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/KeyTransportAlgorithmPredicate.java
@@ -30,7 +30,7 @@ import org.opensaml.security.credential.Credential;
  * An interface for predicates which allow evaluation of a candidate key transport algorithm relative to
  * a candidate data encryption algorithm and/or a candidate key transport credential.
  */
-public interface KeyTransportAlgorithmPredicate extends Predicate<KeyTransportAlgorithmPredicate.SelectionInput> {
+public interface KeyTransportAlgorithmPredicate extends Predicate<KeyTransportAlgorithmPredicate.SelectionInput>, java.io.Serializable {
     
     /**
      * Input class for instances of {@link KeyTransportAlgorithmPredicate}.
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SecurityConfigurationSupport.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SecurityConfigurationSupport.java
index 162f968c1..e604bee73 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SecurityConfigurationSupport.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SecurityConfigurationSupport.java
@@ -22,7 +22,7 @@ import org.opensaml.core.config.ConfigurationService;
 /**
  * Helper methods for working with security configuration.
  */
-public final class SecurityConfigurationSupport {
+public final class SecurityConfigurationSupport implements java.io.Serializable {
     
     /** Constructor. */
     private SecurityConfigurationSupport() { }
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningConfiguration.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningConfiguration.java
index dd350da44..8516a5a63 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningConfiguration.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningConfiguration.java
@@ -34,7 +34,7 @@ import org.opensaml.xmlsec.keyinfo.NamedKeyInfoGeneratorManager;
  * The configuration information to use when generating an XML signature.
  */
 @SuppressWarnings("removal")
-public interface SignatureSigningConfiguration extends WhitelistBlacklistConfiguration {
+public interface SignatureSigningConfiguration extends WhitelistBlacklistConfiguration, java.io.Serializable {
     
     /**
      * Get the list of signing credentials to use when signing, in preference order.
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningParameters.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningParameters.java
index b97bf5e93..9f889c9a7 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningParameters.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningParameters.java
@@ -26,7 +26,7 @@ import org.opensaml.xmlsec.keyinfo.KeyInfoGenerator;
 /**
  * The effective parameters to use when generating an XML signature.
  */
-public class SignatureSigningParameters {
+public class SignatureSigningParameters implements java.io.Serializable {
     
     /** The signing credential. */
     private Credential signingCredential;
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningParametersResolver.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningParametersResolver.java
index fc4f33910..8ccd5b472 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningParametersResolver.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureSigningParametersResolver.java
@@ -23,6 +23,6 @@ import net.shibboleth.utilities.java.support.resolver.Resolver;
 /**
  * An interface for components which resolve {@link SignatureSigningParameters} based on a {@link CriteriaSet}.
  */
-public interface SignatureSigningParametersResolver extends Resolver<SignatureSigningParameters, CriteriaSet> {
+public interface SignatureSigningParametersResolver extends Resolver<SignatureSigningParameters, CriteriaSet>, java.io.Serializable {
 
 }
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationConfiguration.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationConfiguration.java
index 1f7f5daa0..f9373ff1a 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationConfiguration.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationConfiguration.java
@@ -25,7 +25,7 @@ import org.opensaml.xmlsec.signature.support.SignatureTrustEngine;
  * The configuration information to use when validating an XML signature.
  */
 @SuppressWarnings("removal")
-public interface SignatureValidationConfiguration extends WhitelistBlacklistConfiguration {
+public interface SignatureValidationConfiguration extends WhitelistBlacklistConfiguration, java.io.Serializable {
     
     /**
      * Get the signature trust engine to use.
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationParameters.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationParameters.java
index 40f79efd0..1fb31dac0 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationParameters.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationParameters.java
@@ -25,7 +25,7 @@ import org.opensaml.xmlsec.signature.support.SignatureTrustEngine;
  * The effective parameters to use when validating an XML signature.
  */
 @SuppressWarnings("removal")
-public class SignatureValidationParameters extends WhitelistBlacklistParameters {
+public class SignatureValidationParameters extends WhitelistBlacklistParameters implements java.io.Serializable {
     
     /** The signature trust engine to use. */
     @Nullable private SignatureTrustEngine signatureTrustEngine;
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationParametersResolver.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationParametersResolver.java
index 90e812dc1..4e445c7ca 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationParametersResolver.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/SignatureValidationParametersResolver.java
@@ -23,6 +23,6 @@ import net.shibboleth.utilities.java.support.resolver.Resolver;
 /**
  * An interface for components which resolve {@link SignatureValidationParameters} based on a {@link CriteriaSet}.
  */
-public interface SignatureValidationParametersResolver extends Resolver<SignatureValidationParameters, CriteriaSet> {
+public interface SignatureValidationParametersResolver extends Resolver<SignatureValidationParameters, CriteriaSet>, java.io.Serializable {
 
 }
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/WhitelistBlacklistConfiguration.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/WhitelistBlacklistConfiguration.java
index 4bf1acbd5..6bc26cc14 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/WhitelistBlacklistConfiguration.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/WhitelistBlacklistConfiguration.java
@@ -33,7 +33,7 @@ import net.shibboleth.utilities.java.support.annotation.constraint.Unmodifiable;
  * @deprecated
  */
 @Deprecated(forRemoval=true, since="4.1.0")
-public interface WhitelistBlacklistConfiguration extends AlgorithmPolicyConfiguration {
+public interface WhitelistBlacklistConfiguration extends AlgorithmPolicyConfiguration, java.io.Serializable {
 
     /** Whitelist/blacklist precedence values. */
     public enum Precedence {
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/WhitelistBlacklistParameters.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/WhitelistBlacklistParameters.java
index de19330a9..3f93b90b5 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/WhitelistBlacklistParameters.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/WhitelistBlacklistParameters.java
@@ -36,7 +36,7 @@ import net.shibboleth.utilities.java.support.primitive.DeprecationSupport.Object
  * @deprecated
  */
 @Deprecated(forRemoval=true, since="4.1.0")
-public class WhitelistBlacklistParameters extends AlgorithmPolicyParameters {
+public class WhitelistBlacklistParameters extends AlgorithmPolicyParameters implements java.io.Serializable {
     
     /**
      * Get the list of whitelisted algorithm URI's.
diff --git a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/keyinfo/KeyInfoGeneratorFactory.java b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/keyinfo/KeyInfoGeneratorFactory.java
index 11d581f68..96550baef 100644
--- a/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/keyinfo/KeyInfoGeneratorFactory.java
+++ b/opensaml-xmlsec-api/src/main/java/org/opensaml/xmlsec/keyinfo/KeyInfoGeneratorFactory.java
@@ -28,7 +28,7 @@ import org.opensaml.xmlsec.signature.KeyInfo;
 /**
  * Interface for factories which produce {@link KeyInfoGenerator} instances.
  */
-public interface KeyInfoGeneratorFactory {
+public interface KeyInfoGeneratorFactory extends java.io.Serializable {
     
     /**
      * Get a new instance of the generator type produced by the factory.
diff --git a/opensaml-xmlsec-impl/src/main/java/org/opensaml/xmlsec/keyinfo/impl/X509KeyInfoGeneratorFactory.java b/opensaml-xmlsec-impl/src/main/java/org/opensaml/xmlsec/keyinfo/impl/X509KeyInfoGeneratorFactory.java
index f452dbf24..f6aa1a13e 100644
--- a/opensaml-xmlsec-impl/src/main/java/org/opensaml/xmlsec/keyinfo/impl/X509KeyInfoGeneratorFactory.java
+++ b/opensaml-xmlsec-impl/src/main/java/org/opensaml/xmlsec/keyinfo/impl/X509KeyInfoGeneratorFactory.java
@@ -59,7 +59,7 @@ import net.shibboleth.utilities.java.support.logic.Constraint;
  * {@link InternalX500DNHandler}. The default output format for subject and issuer DN's is RFC2253.
  * The default set of subject alternative names to process is empty.
  */
-public class X509KeyInfoGeneratorFactory extends BasicKeyInfoGeneratorFactory {
+public class X509KeyInfoGeneratorFactory extends BasicKeyInfoGeneratorFactory implements java.io.Serializable {
     
     /** The set of options configured for the factory. */
     private final X509Options options;
@@ -391,7 +391,7 @@ public class X509KeyInfoGeneratorFactory extends BasicKeyInfoGeneratorFactory {
      * An implementation of {@link KeyInfoGenerator} capable of handling the information 
      * contained within a {@link X509Credential}.
      */
-    public class X509KeyInfoGenerator extends BasicKeyInfoGenerator {
+    public class X509KeyInfoGenerator extends BasicKeyInfoGenerator implements java.io.Serializable {
 
         /** Class logger. */
         private final Logger log = LoggerFactory.getLogger(X509KeyInfoGenerator.class);
@@ -736,7 +736,7 @@ public class X509KeyInfoGeneratorFactory extends BasicKeyInfoGeneratorFactory {
     /**
     * Options to be used in the production of a {@link KeyInfo} from an {@link X509Credential}.
     */
-   protected class X509Options extends BasicOptions {
+   protected class X509Options extends BasicOptions implements java.io.Serializable {
        
        /** Emit the entity certificate as an X509Certificate element within X509Data. */
        private boolean emitEntityCertificate;
