diff --git a/pom.xml b/pom.xml
index f91007f..f4f3570 100644
--- a/pom.xml
+++ b/pom.xml
@@ -3,13 +3,13 @@
     xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
 
     <modelVersion>4.0.0</modelVersion>
-
+<!--
     <parent>
         <groupId>net.shibboleth</groupId>
         <artifactId>parent</artifactId>
         <version>11.3.5</version>
     </parent>
-
+-->
     <name>java-support</name>
     <groupId>net.shibboleth.utilities</groupId>
     <artifactId>java-support</artifactId>
@@ -22,8 +22,35 @@
 
     <properties>
         <automatic.module.name>net.shibboleth.utilities.java.support</automatic.module.name>
+        <slf4j.groupId>org.slf4j</slf4j.groupId>
+    	<nashorn.jdk.version>15.3</nashorn.jdk.version>
+
+		<maven.compiler.source>11</maven.compiler.source>
+		<maven.compiler.target>11</maven.compiler.target>
     </properties>
 
+	<dependencyManagement>
+		<dependencies>
+            		<dependency>
+			        <groupId>net.shibboleth</groupId>
+			        <artifactId>parent</artifactId>
+			        <version>11.3.5</version>
+		                <scope>import</scope>
+		                <type>pom</type>
+            		</dependency>
+		</dependencies>
+	</dependencyManagement>
+
+	<repositories>
+		<repository>
+			<id>shib</id>
+			<name>Shib</name>
+			<url>https://build.shibboleth.net/nexus/content/repositories/releases/</url>
+			<releases></releases>
+			<snapshots></snapshots>
+		</repository>
+	</repositories>
+
     <scm>
         <connection>scm:git:https://git.shibboleth.net/git/${project.artifactId}</connection>
         <developerConnection>scm:git:git@git.shibboleth.net:${project.artifactId}</developerConnection>
@@ -117,6 +144,7 @@
         <!-- Runtime Dependencies -->
 
         <!-- Test Dependencies -->
+<!--
         <dependency>
             <groupId>org.testng</groupId>
             <artifactId>testng</artifactId>
@@ -137,7 +165,7 @@
             <artifactId>spring-web</artifactId>
             <scope>test</scope>
         </dependency>
-
+-->
     </dependencies>
 
     <profiles>
diff --git a/src/main/java/net/shibboleth/utilities/java/support/collection/ClassIndexedSet.java b/src/main/java/net/shibboleth/utilities/java/support/collection/ClassIndexedSet.java
index 7b3fbdd..98e217f 100644
--- a/src/main/java/net/shibboleth/utilities/java/support/collection/ClassIndexedSet.java
+++ b/src/main/java/net/shibboleth/utilities/java/support/collection/ClassIndexedSet.java
@@ -34,7 +34,7 @@ import net.shibboleth.utilities.java.support.logic.Constraint;
  * 
  * @param <T> the type of object stored by this class
  */
-public class ClassIndexedSet<T> extends AbstractSet<T> implements Set<T> {
+public class ClassIndexedSet<T> extends AbstractSet<T> implements Set<T>, java.io.Serializable {
 
     /** Storage for set members. */
     private final HashSet<T> set;
diff --git a/src/main/java/net/shibboleth/utilities/java/support/collection/ClassToInstanceMultiMap.java b/src/main/java/net/shibboleth/utilities/java/support/collection/ClassToInstanceMultiMap.java
index 54cdf78..35a9a81 100644
--- a/src/main/java/net/shibboleth/utilities/java/support/collection/ClassToInstanceMultiMap.java
+++ b/src/main/java/net/shibboleth/utilities/java/support/collection/ClassToInstanceMultiMap.java
@@ -44,7 +44,7 @@ import javax.annotation.concurrent.NotThreadSafe;
  * @param <B> a bound for the types of values in the map
  */
 @NotThreadSafe
-public class ClassToInstanceMultiMap<B> {
+public class ClassToInstanceMultiMap<B> implements java.io.Serializable {
 
     /** Whether supertypes should also be indexed. */
     private final boolean indexSupertypes;
diff --git a/src/main/java/net/shibboleth/utilities/java/support/component/AbstractIdentifiableInitializableComponent.java b/src/main/java/net/shibboleth/utilities/java/support/component/AbstractIdentifiableInitializableComponent.java
index 6f4f26a..b5e432d 100644
--- a/src/main/java/net/shibboleth/utilities/java/support/component/AbstractIdentifiableInitializableComponent.java
+++ b/src/main/java/net/shibboleth/utilities/java/support/component/AbstractIdentifiableInitializableComponent.java
@@ -27,7 +27,7 @@ import net.shibboleth.utilities.java.support.annotation.constraint.NotEmpty;
  */
 @ThreadSafe
 public abstract class AbstractIdentifiableInitializableComponent extends AbstractIdentifiedInitializableComponent
-        implements IdentifiableComponent {
+        implements IdentifiableComponent, java.io.Serializable {
 
     /** {@inheritDoc} */
     @Override public synchronized void setId(@Nonnull @NotEmpty final String componentId) {
diff --git a/src/main/java/net/shibboleth/utilities/java/support/component/Component.java b/src/main/java/net/shibboleth/utilities/java/support/component/Component.java
index c5b0170..1ffac53 100644
--- a/src/main/java/net/shibboleth/utilities/java/support/component/Component.java
+++ b/src/main/java/net/shibboleth/utilities/java/support/component/Component.java
@@ -18,6 +18,6 @@
 package net.shibboleth.utilities.java.support.component;
 
 /** A marker interface that represents something used as a discrete part of a larger system. */
-public interface Component {
+public interface Component extends java.io.Serializable {
 
 }
\ No newline at end of file
diff --git a/src/main/java/net/shibboleth/utilities/java/support/security/DataSealer.java b/src/main/java/net/shibboleth/utilities/java/support/security/DataSealer.java
index ca63b5f..05812aa 100644
--- a/src/main/java/net/shibboleth/utilities/java/support/security/DataSealer.java
+++ b/src/main/java/net/shibboleth/utilities/java/support/security/DataSealer.java
@@ -81,7 +81,7 @@ public class DataSealer extends AbstractInitializableComponent {
     @NonnullAfterInit private DataSealerKeyStrategy keyStrategy;
 
     /** Source of secure random data. */
-    @NonnullAfterInit private SecureRandom random;
+    @NonnullAfterInit private transient SecureRandom random;
 
     /** Encodes encrypted bytes to string. */
     @Nonnull private BinaryEncoder encoder;
diff --git a/src/main/java/net/shibboleth/utilities/java/support/security/impl/RandomIdentifierGenerationStrategy.java b/src/main/java/net/shibboleth/utilities/java/support/security/impl/RandomIdentifierGenerationStrategy.java
index 7d03503..447d3e2 100644
--- a/src/main/java/net/shibboleth/utilities/java/support/security/impl/RandomIdentifierGenerationStrategy.java
+++ b/src/main/java/net/shibboleth/utilities/java/support/security/impl/RandomIdentifierGenerationStrategy.java
@@ -38,7 +38,7 @@ import org.apache.commons.codec.binary.StringUtils;
 public class RandomIdentifierGenerationStrategy implements IdentifierGenerationStrategy {
 
     /** Random number generator. */
-    private final Random random;
+    private final transient Random random;
 
     /** Number of random bytes in the identifier. */
     private final int sizeOfIdentifier;
